### ISSUE TITLE
A small problem about time slices when using inoisy files

### ISSUE THREAD

**Issue by zhenyuzhang98** [iAART/aart#2]
Excellent work, really helpful for my research! 

Here is my question:

In some code using inoisy files, for example, line 86 of iMovies.py

times = np.linspace(xtstart, xtend, nt)

Should that be np.arange instead of np.linspace?

It seems that using np.linspace will change dx0.

**Comment by alejandroc137** [comment]
Hi @zhenyuzhang98. Thanks for the question and I am glad to read you find AART useful. 

The notation here is indeed confusing:

`times = np.linspace(xtstart, xtend, nt) `

is meant to provide the time steps at which the inoisy files are stored, so, e.g., if you have a movie that started at `xtstart=0` and when up to `xtend=1000` and you have `nt=10` snapshots (these are measured in units of M), it means you have the movie at these `times=array([   0.  ,  111.11111111, ..., 1000. ])`, i.e., a cadence of ~111M. 

That cadence is not the parameter _dx0_ that appears in `params.py`. There, `dxn`, means the resolution of the spatial grid for the nth lensing band, which sets the resolution for the raytracing.

**Comment by zhenyuzhang98** [comment]
Thank you for your kind reply! However, it is not my question.

I mean the dx0 from inoisy output, namely, inoisy.h5/params/dx0 (which is not included in your example inoisy file but is common in files genrated from inoisy).

To my understand, for example, if I have an inoisy movie that started at xtstart=0 and when up to xtend=1000 and I have nt=10 snapshots, it means the time of the inoisy file is [ 0., 100., 200., ..., 900. ]. And the t=1000 is the same as t=0 because of the periodic boundary conditions. 

But if we use np.linspace then it will become array([   0.  ,  111.11111111, ..., 1000. ], which may not correspond to inoisy output correctly. Then the interpolation in line 107 of iMovies.py will have bias.

**Comment by alejandroc137** [comment]
Oh, I see your point. What I wrote before was misleading. I now remember, I think, what we were doing actually there. For the interpolation, we needed to copy the periodic boundary. That is why in, e.g., line 107 of iMovies.py we are doing this to the source data:

data=np.concatenate((data,data[0,:,:][np.newaxis,:,:]),axis=0)

So, nt = data.shape[0] is now the tsize+1, so when we do 

times = np.linspace(xtstart, xtend, nt)

In the example above, the times array will actually be [   0.,  100.,  200., ..., 1000.], and the cadence will now coincide with inoisy.h5/params/dx0 (which we are not storing in that example).
